### PR TITLE
Feature/ft 5 axios

### PR DESCRIPTION
**Pull request by The-Luc**
- Setup axios instance and interceptor
- Modularized store

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-  <div id="app"></div>
+  <div id="app">
+    App Home Page
+  </div>
 </template>
 
+<script>
+export default {};
+</script>
+
```
Comment (trantiendathcmnre) [inline src/App.vue:10]: Please check and remove if we don't need this code

File: src/store/auth.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  signin({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
+        saveToken(res.data.token);
+        commit('setAuthState', true);
+      })
+      .catch(error => console.log('Sign in ERROR --- ' + error));
+  },
```
Comment (Koo-SSS) [inline src/store/auth.js:31]: 1/ Should create service file to handle call API.
 - Create folder services and then create file {module}.js to handle call API. Example services/book.js, services/auth.js
 - And in action, import service to use
2/ Should try async/await

File: src/store/auth.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  // SIGN UP user and save token
+  signup({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
```
Comment (trantiendathcmnre) [inline src/store/auth.js]: we should be check response data be for do any action to make sure properties of object exist
if (res) {
}

File: src/store/auth.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+const actions = {
+  // sign in user and save token
+  signin({ commit }, user) {
+    api
+      .post('/users/signin', user)
+      .then(res => {
+        console.log(res.data.message);
+        saveToken(res.data.token);
```
Comment (trantiendathcmnre) [inline src/store/auth.js]: we should be check response data be for do any action to make sure properties of object exist
if (res) {
}

File: src/store/book.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  setBooks: (state, books) => (state.books = books)
+};
+
+const actions = {
+  // get BOOK data from API
+  fetchBook({ commit }) {
+    api
+      .get('/books')
+      .then(res => {
+        commit('setBooks', res.data);
```
Comment (trantiendathcmnre) [inline src/store/book.js]: we should be check response data be for do any action to make sure properties of object exist
if (res) {
}

**Review by Koo-SSS — APPROVED** [review]
LGTM

File: src/store/category.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  setCategory: (state, data) => (state.category = data)
+};
+
+const actions = {
+  // get CATEGORY data from API
+  fetchCategory({ commit }) {
+    api
+      .get('/category')
+      .then(res => {
+        commit('setCategory', res.data);
```
Comment (trantiendathcmnre) [inline src/store/category.js]: we should be check response data be for do any action to make sure properties of object exist
if (res) {
}

File: src/App.vue (hunk 1 above)
Comment (The-Luc) [reply src/App.vue:10]: That tag will be used soon when UI components added.

Comment (trantiendathcmnre) [reply src/App.vue:10]: okay a @The-Luc

File: src/store/auth.js (hunk 2 above)
Comment (The-Luc) [reply src/store/auth.js:31]: Done

**Pull request merged**